libs/core/langchain_core/messages/utils.py

```diff
@@ -845,6 +862,775 @@ def list_token_counter(messages: Sequence[BaseMessage]) -> int:
         )
 
 
+def _runnable_generator(func: Callable) -> Callable:
+    @overload
+    def wrapped(
+        messages: Literal[None] = None, **kwargs: Any
+    ) -> Runnable[
+        Union[MessageLikeRepresentation, Sequence[MessageLikeRepresentation]],
+        Union[BaseMessage, List[BaseMessage]],
+    ]: ...
+
+    @overload
+    def wrapped(
+        messages: Sequence[Union[BaseMessage, Dict, Tuple]], **kwargs: Any
+    ) -> List[BaseMessage]: ...
+
+    @overload
+    def wrapped(messages: MessageLikeRepresentation, **kwargs: Any) -> BaseMessage: ...
+
+    def wrapped(
+        messages: Union[
+            MessageLikeRepresentation, Sequence[MessageLikeRepresentation], None
+        ] = None,
+        **kwargs: Any,
+    ) -> Union[
+        BaseMessage,
+        List[BaseMessage],
+        Runnable[
+            Union[MessageLikeRepresentation, Sequence[MessageLikeRepresentation]],
+            Union[BaseMessage, List[BaseMessage]],
+        ],
+    ]:
+        from langchain_core.runnables.base import RunnableGenerator
+
+        if messages is not None:
+            return func(messages, **kwargs)
+        else:
+
+            def transform(input_: Iterator, **kwargs: Any) -> Iterator:
+                block_indexes = set()
+                for x in input_:
+                    msg = func(x, **kwargs)
+                    # Special handling for transforming an OpenAI stream to an
+                    # Anthropic stream.
+                    if isinstance(msg, AIMessageChunk) and isinstance(
+                        msg.content, list
+                    ):
+                        tool_use_ct = 0
+                        for block in msg.content:
+                            if not isinstance(block, dict):
+                                continue
+                            if "index" in block:
+                                block_indexes.add(block["index"])
+                            elif block.get("type") == "tool_use":
+                                block["index"] = max(len(block_indexes), *block_indexes)
+                                msg.tool_call_chunks[tool_use_ct]["index"] = block[
+                                    "index"
+                                ]
+                            else:
+                                pass
+
+                            if block.get("type") == "tool_use":
+                                tool_use_ct += 1
+                    else:
+                        block_indexes.add(0)
+                    yield msg
+
+            return RunnableGenerator(partial(transform, **kwargs), name=func.__name__)
+
+    wrapped.__doc__ = func.__doc__
+    return wrapped
+
+
+@_runnable_generator
+def format_messages_as(
```

Comment:
```suggestion
def format_messages(
```
? 🌝

Reply:
don't we also have a `convert_messages_to_dict` somewhere that does the same thing? Would that kind of naming make more sense?

Might make it clearer this is only meant to handle conversion to dicts (incase anyone tries to add conversions to custom sdks at some point)

Reply:
watcha mean meant to handle conversion to dict?

Reply:
it formats the contents as openai/anthropic dicts, not <insert some model provider sdk> basemodel/protobuf/etc types for theirs.